### DISCUSSION TITLE
fastapi Unable to get request parameters in request

### DISCUSSION THREAD

**Discussion by chenchen1992223** [fastapi/fastapi#6593] [Questions] [answered]
### First Check

- [X] I added a very descriptive title to this issue.
- [X] I used the GitHub search to find a similar issue and didn't find it.
- [X] I searched the FastAPI documentation, with the integrated search.
- [X] I already searched in Google "How to X in FastAPI" and didn't find any information.
- [X] I already read and followed all the tutorial in the docs and didn't find an answer.
- [X] I already checked if it is not related to FastAPI but to [Pydantic](https://github.com/samuelcolvin/pydantic).
- [X] I already checked if it is not related to FastAPI but to [Swagger UI](https://github.com/swagger-api/swagger-ui).
- [X] I already checked if it is not related to FastAPI but to [ReDoc](https://github.com/Redocly/redoc).

### Commit to Help

- [X] I commit to help with one of those options 👆

### Example Code

```python
test3.py
from pydantic import BaseModel
from fastapi import APIRouter
class CreateUser(BaseModel):
    username: str
    password: str
test=APIRouter()
@test.post("login")
def login(login:CreateUser):
    return login
test2.py
import uvicorn

from fastapi import FastAPI,Request
from test3 import test
import time

app = FastAPI()
@app.middleware('http')
async def add_process_time_header(request: Request, call_next): # call_next将接收request请求做为参数
    start_time = time.time()
    query_param=request.query_params
    print("query_param is null={}",query_param)
    response = await call_next(request)
    return response

app.include_router(router=test,prefix='/login', tags=['分析模块'])

if __name__ == '__main__':
    uvicorn.run(app='test2:app', host="0.0.0.0", port=8002, reload=True, debug=True)
```

### Description

My request class inherits pydantic's basemodel, and then uses middleware to obtain the request parameters when requesting. It is found that the request parameters are empty and query_ Params method, I hope you can answer

### Operating System

macOS

### FastAPI Version

 0.63.0

### Python Version

3.8

**Comment by ghandic** [comment]
Pydantic base models will go into the body fo the request - you could you check the swagger docs the app creates to confirm what the curl command it generates is?

**Comment by chenchen1992223** [comment]
> Pydantic base models will go into the body fo the request - you could you check the swagger docs the app creates to confirm what the curl command it generates is

I checked that there is no request body. You can see the example I wrote. In this example, no parameters are obtained,

**Comment by ghandic** [comment, marked as answer]
```python
@app.middleware("http")
async def add_process_time_header(request: Request, call_next):  # call_next将接收request请求做为参数
    start_time = time.time()
    query_param = request.query_params
    print(await request.json())
    print("query_param is null={}", query_param)
    response = await call_next(request)
    return response
```

It is in the body - see this

**Comment by ghandic** [comment]
`await request.json()` will parse the body and that's where the parameters are since it is a `pydandic` model it will go into the body - you can use `deepObject` if you prefer. This isn't currently supported on FastAPI but I have an implementation here https://github.com/ghandic/FastAPI-deepObject

**Comment by ghandic** [comment]
@chenchen1992223 did this answer your question? If so could you close the issue 😄

**Comment by chenchen1992223** [comment]
Thank you for your answer. Thank you.
> @chenchen1992223 did this answer your question? If so could you close the issue 😄
